protocol/contracts/amo/helpers/BalancerPoolHelper.sol

```diff
@@ -0,0 +1,428 @@
+pragma solidity ^0.8.17;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (amo/helpers/BalancerPoolHelper.sol)
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { IBalancerPoolHelper } from "contracts/interfaces/amo/helpers/IBalancerPoolHelper.sol";
+import { IBalancerVault } from "contracts/interfaces/external/balancer/IBalancerVault.sol";
+import { IBalancerHelpers } from "contracts/interfaces/external/balancer/IBalancerHelpers.sol";
+import { AMOCommon } from "contracts/amo/helpers/AMOCommon.sol";
+
+interface IWeightPool2Tokens {
+    function getNormalizedWeights() external view returns (uint256[] memory);
+}
+
+contract BalancerPoolHelper is IBalancerPoolHelper {
+    using SafeERC20 for IERC20;
+
+    IBalancerVault public immutable override balancerVault;
+    IBalancerHelpers public immutable override balancerHelpers;
+    IERC20 public immutable override bptToken;
+    IERC20 public immutable override protocolToken;
+    IERC20 public immutable override quoteToken;
+    address public immutable override amo;
+    
+    uint256 public constant override BPS_PRECISION = 10_000;
+    uint256 public constant override PRICE_PRECISION = 1e18;
+
+    // @notice protocolToken index in balancer pool
+    uint64 public immutable override protocolTokenIndexInBalancerPool;
```

Comment:
I just learnt something new. For some reason, I had it in my head that byte packing only happened to structs, rather than on top level primitives too.

You're right!

Reply:
Done in https://github.com/TempleDAO/temple/pull/814/files